jwst/regtest/st_fitsdiff.py

```diff
@@ -1268,57 +1357,104 @@ def _diff(self):
-                    else:
-                        self.total_diff_per_col[col.name] += len(set(diffs[0]))
+                self.diff_total += len(set(diffs[0]))
 
-            # The following lines are identical to the original TableDataDiff code
+                if self.numdiffs >= 0:
+                    if len(self.diff_values) >= self.numdiffs:
+                        # Don't save any more diff values
+                        continue
 
-            if self.numdiffs >= 0:
-                if len(self.diff_values) >= self.numdiffs:
-                    # Don't save any more diff values
-                    continue
+                    # Add no more diff'd values than this
+                    max_diffs = self.numdiffs - len(self.diff_values)
+                else:
+                    max_diffs = len(diffs[0])
+
+                last_seen_idx = None
+                for idx in islice(diffs[0], 0, max_diffs):
+                    if idx == last_seen_idx:
+                        # Skip duplicate indices, which my occur when the column
+                        # data contains multi-dimensional values; we're only
+                        # interested in storing row-by-row differences
+                        continue
+                    last_seen_idx = idx
+                    self.diff_values.append(((col.name, idx), (arra[idx], arrb[idx])))
 
-                # Add no more diff'd values than this
-                max_diffs = self.numdiffs - len(self.diff_values)
             else:
-                max_diffs = len(diffs[0])
-
-            last_seen_idx = None
-            for idx in islice(diffs[0], 0, max_diffs):
-                if idx == last_seen_idx:
-                    # Skip duplicate indices, which my occur when the column
-                    # data contains multi-dimensional values; we're only
-                    # interested in storing row-by-row differences
-                    continue
-                last_seen_idx = idx
-                self.diff_values.append(((col.name, idx), (arra[idx], arrb[idx])))
+                # The following lines include STScI's changes:
+                # - Calculate the absolute and relative differences separately for ad hoc report
+                # - If report_pixel_loc_diffs is False, just get the total differences,
+                #   it is not important where they come from
+
+                # Calculate the absolute and relative differences
+                nan_idx = np.isnan(arra) | np.isnan(arrb)
+                anonan = arra[~nan_idx]
+                bnonan = arrb[~nan_idx]
+                diffs = np.abs(anonan - bnonan)
+                abs_diffs = diffs[diffs > self.atol].size
+                nozeros = (diffs != 0.0) & (bnonan != 0.0)
+                rel_values = diffs[nozeros] / np.abs(bnonan[nozeros])
+                rel_diffs = rel_values[rel_values > self.rtol].size
+
+                if abs_diffs > 0 or rel_diffs > 0:
```

Comment:
I think this + the next suggestion is why it's showing headers but no reports for table extensions.  This is a different criterion than is used in the usual diff -- it needs to check on atol + rtol * abs(b) before deciding whether to report a difference.

Something like this?
```suggestion
                sum_diffs = np.any(diffs > (self.atol + self.rtol * np.abs(bnonan)))
                if sum_diffs:
```